OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -58,6 +58,29 @@ public static void ProcessOrder(OrderManager orderManager, World world, int clie
 						break;
 					}
 
+				case "PlaceBeacon":
+					{
+						var client = orderManager.LobbyInfo.ClientWithIndex(clientId);
+						if (client != null)
+						{
+							var player = world.FindPlayerByClient(client);
+							if (player == null)
+							{
+								if (orderManager.LocalClient.IsObserver)
+								{
+									Game.AddChatLine(Color.White, ServerChatName, "{0} placed a beacon".F(client.Name));
+									player = world.Players[0];
```

Comment:
Yes, I'm not sure if there's a cleaner way to get it, but would `player = world.Players.Single((p) => p.PlayerName == "Everyone");` work better? It's at least more clear what is going on then, as long as the name doesn't change. It's set [here ](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs#L64).